### PR TITLE
ec2 resources -  tag on create

### LINKED ISSUE #14011: Various EC2 resources: Tag-on-create
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

##### 2020-06-30

> Added support for tag-on-create for CreateVpc, CreateEgressOnlyInternetGateway, CreateSecurityGroup, CreateSubnet, CreateNetworkInterface, CreateNetworkAcl, CreateDhcpOptions and CreateInternetGateway. You can now specify tags when creating any of these resources.

##### 2020-07-20

> Added support for tag-on-create for CreateVpcPeeringConnection and CreateRouteTable. You can now specify tags when creating any of these resources.

##### 2020-07-28

> Introduces support for tag-on-create capability for the following APIs: CreateVpnConnection, CreateVpnGateway, and CreateCustomerGateway.

##### 2020-06-30

* `aws_vpc`
* `aws_egress_only_internet_gateway`
* `aws_security_group`
* `aws_subnet`
* `aws_network_interface`
* `aws_network_acl`
* `aws_vpc_dhcp_options`
* `aws_internet_gateway`

##### 2020-07-20

* `aws_route_table`
* `aws_vpc_peering_connection`
* `aws_vpc_peering_connection_accepter`

##### 2020-07-28

* `aws_customer_gateway`
* `aws_vpn_connection`
* `aws_vpn_gateway`

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/07/amazon-vpc-resources-support-tag-on-create/).
[Announcement for Site-to-Site VPN resources](https://aws.amazon.com/about-aws/whats-new/2020/07/aws-site-to-site-vpn-supports-tagging-resources-creation-resource-level/).

##### 2020-06-30

Requires [AWS SDK v1.32.13](https://github.com/aws/aws-sdk-go/releases/tag/v1.32.13):

* https://github.com/terraform-providers/terraform-provider-aws/pull/14000

##### 2020-07-20

Requires [AWS SDK v1.33.8](https://github.com/aws/aws-sdk-go/releases/tag/v1.33.8):

* https://github.com/terraform-providers/terraform-provider-aws/pull/14000

##### 2020-07-28

Requires [AWS SDK v1.33.14](https://github.com/aws/aws-sdk-go/releases/tag/v1.33.14):

* https://github.com/terraform-providers/terraform-provider-aws/pull/14000

### LINKED ISSUE #14114: r/aws_spot_instance_request: Tag-on-create
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

> EC2 Spot now enables customers to tag their Spot Instances Requests on creation.

### New or Affected Resource(s)

* [`aws_spot_instance_request`](https://www.terraform.io/docs/providers/aws/r/spot_instance_request.html)

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/07/amazon-ec2-now-supports-tagging-ec2-spot-instance-requests/).

Requires [AWS SDK v1.33.4](https://github.com/aws/aws-sdk-go/releases/tag/v1.33.4):
* #14000

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #14114
Closes #14011

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_customer_gateway - support tag on create
resource_aws_network_interface - support tag on create 
resource_aws_security_group - support tag on create 
resource_aws_route_table - support tag on create 
resource_aws_vpn_connection - support tag on create 
resource_aws_vpc_peering_connection - support tag on create 
resource_aws_dhcp_options - support tag on create 
resource_aws_vpn_gateway -  support tag on create 
resource_aws_spot_instance_request - support tag on create 
resource_aws_intrenet_gateway - support tag on create 
resource_aws_egress_only_intrenet_gateway  - support tag on create 
resource_aws_network_acl - support tag on create 
resource_aws_subnet - support tag on create 
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'
--- PASS: TestAccAWSVpc_basic (67.54s)
--- PASS: TestAccAWSVpc_tags (199.74s)

--- PASS: TestAccAWSCustomerGateway_basic (149.96s)
--- PASS: TestAccAWSCustomerGateway_tags (107.17s)

--- PASS: TestAccAWSVpnConnection_basic (735.95s)
--- PASS: TestAccAWSVpnConnection_tags (440.83s)

--- PASS: TestAccAWSDHCPOptions_basic (41.26s)
--- PASS: TestAccAWSDHCPOptions_tags (82.67s)

--- PASS: TestAccAWSSpotInstanceRequest_basic (109.74s)
--- PASS: TestAccAWSSpotInstanceRequest_tags (180.10s)

--- PASS: TestAccAWSVpnGateway_basic (168.59s)
--- PASS: TestAccAWSVpnGateway_tags (173.47s)

--- PASS: TestAccAWSRouteTable_basic (144.60s)
--- PASS: TestAccAWSRouteTable_tags (150.90s)

--- PASS: TestAccAWSSecurityGroup_tags (150.92s)
--- PASS: TestAccAWSSecurityGroup_basic (75.34s)

--- PASS: TestAccAWSNetworkAcl_basic (94.25s)
--- PASS: TestAccAWSNetworkAcl_tags (154.71s)

--- PASS: TestAccAWSInternetGateway_basic (167.59s)
--- PASS: TestAccAWSInternetGateway_tags (169.32s)

--- PASS: TestAccAWSEgressOnlyInternetGateway_basic (69.70s)
--- PASS: TestAccAWSEgressOnlyInternetGateway_Tags (187.81s)

--- PASS: TestAccAWSVPCPeeringConnection_basic (78.24s)
--- PASS: TestAccAWSVPCPeeringConnection_tags (168.22s)

--- PASS: TestAccAWSENI_tags (219.87s)
--- PASS: TestAccAWSENI_basic (86.60s)

--- PASS: TestAccAWSSubnet_basic (87.13s)
--- PASS: TestAccAWSSubnet_tags (193.35s)
```

### CODE REVIEW

File: aws/resource_aws_network_interface_test.go [hunk 1]
```diff
@@ -832,3 +871,124 @@ resource "aws_network_interface" "test" {
...
+}
+
+data "aws_availability_zones" "available" {
+   state = "available"
+
+   filter {
+     name   = "opt-in-status"
+     values = ["opt-in-not-required"]
+   }
+ }
```
Comment (ewbankkit) [inline aws/resource_aws_network_interface_test.go:893]: Prefer `testAccAvailableAZsNoOptInConfig()`: #14500.

File: aws/resource_aws_network_interface_test.go [hunk 2]
```diff
@@ -832,3 +871,124 @@ resource "aws_network_interface" "test" {
...
```
Comment (ewbankkit) [inline aws/resource_aws_network_interface_test.go:953]: Prefer `testAccAvailableAZsNoOptInConfig()`: #14500.

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSVpc_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSVpc_basic -timeout 120m
=== RUN   TestAccAWSVpc_basic
=== PAUSE TestAccAWSVpc_basic
=== CONT  TestAccAWSVpc_basic
--- PASS: TestAccAWSVpc_basic (26.16s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	26.202s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSVpc_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSVpc_tags -timeout 120m
=== RUN   TestAccAWSVpc_tags
=== PAUSE TestAccAWSVpc_tags
=== CONT  TestAccAWSVpc_tags
--- PASS: TestAccAWSVpc_tags (61.65s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	61.697s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSCustomerGateway_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSCustomerGateway_basic -timeout 120m
=== RUN   TestAccAWSCustomerGateway_basic
=== PAUSE TestAccAWSCustomerGateway_basic
=== CONT  TestAccAWSCustomerGateway_basic
--- PASS: TestAccAWSCustomerGateway_basic (43.27s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	43.315s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSCustomerGateway_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSCustomerGateway_tags -timeout 120m
=== RUN   TestAccAWSCustomerGateway_tags
=== PAUSE TestAccAWSCustomerGateway_tags
=== CONT  TestAccAWSCustomerGateway_tags
--- PASS: TestAccAWSCustomerGateway_tags (39.46s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	39.511s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSVpnConnection_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSVpnConnection_basic -timeout 120m
=== RUN   TestAccAWSVpnConnection_basic
=== PAUSE TestAccAWSVpnConnection_basic
=== CONT  TestAccAWSVpnConnection_basic
--- PASS: TestAccAWSVpnConnection_basic (741.70s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	741.758s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSVpnConnection_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSVpnConnection_tags -timeout 120m
=== RUN   TestAccAWSVpnConnection_tags
=== PAUSE TestAccAWSVpnConnection_tags
=== CONT  TestAccAWSVpnConnection_tags
--- PASS: TestAccAWSVpnConnection_tags (487.77s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	487.814s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSDHCPOptions_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSDHCPOptions_basic -timeout 120m
=== RUN   TestAccAWSDHCPOptions_basic
=== PAUSE TestAccAWSDHCPOptions_basic
=== CONT  TestAccAWSDHCPOptions_basic
--- PASS: TestAccAWSDHCPOptions_basic (13.75s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	13.800s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSDHCPOptions_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSDHCPOptions_tags -timeout 120m
=== RUN   TestAccAWSDHCPOptions_tags
=== PAUSE TestAccAWSDHCPOptions_tags
=== CONT  TestAccAWSDHCPOptions_tags
--- PASS: TestAccAWSDHCPOptions_tags (29.27s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	29.341s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSSpotInstanceRequest_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSSpotInstanceRequest_basic -timeout 120m
=== RUN   TestAccAWSSpotInstanceRequest_basic
=== PAUSE TestAccAWSSpotInstanceRequest_basic
=== CONT  TestAccAWSSpotInstanceRequest_basic
--- PASS: TestAccAWSSpotInstanceRequest_basic (294.19s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	294.262s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSSpotInstanceRequest_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSSpotInstanceRequest_tags -timeout 120m
=== RUN   TestAccAWSSpotInstanceRequest_tags
=== PAUSE TestAccAWSSpotInstanceRequest_tags
=== CONT  TestAccAWSSpotInstanceRequest_tags
--- PASS: TestAccAWSSpotInstanceRequest_tags (79.09s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	79.133s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSVpnGateway_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSVpnGateway_basic -timeout 120m
=== RUN   TestAccAWSVpnGateway_basic
=== PAUSE TestAccAWSVpnGateway_basic
=== CONT  TestAccAWSVpnGateway_basic
--- PASS: TestAccAWSVpnGateway_basic (91.59s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	91.648s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSVpnGateway_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSVpnGateway_tags -timeout 120m
=== RUN   TestAccAWSVpnGateway_tags
=== PAUSE TestAccAWSVpnGateway_tags
=== CONT  TestAccAWSVpnGateway_tags
--- PASS: TestAccAWSVpnGateway_tags (77.40s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	77.447s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSRouteTable_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSRouteTable_basic -timeout 120m
=== RUN   TestAccAWSRouteTable_basic
=== PAUSE TestAccAWSRouteTable_basic
=== CONT  TestAccAWSRouteTable_basic
--- PASS: TestAccAWSRouteTable_basic (60.55s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	60.616s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSRouteTable_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSRouteTable_tags -timeout 120m
=== RUN   TestAccAWSRouteTable_tags
=== PAUSE TestAccAWSRouteTable_tags
=== CONT  TestAccAWSRouteTable_tags
--- PASS: TestAccAWSRouteTable_tags (57.97s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	58.028s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSSecurityGroup_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSSecurityGroup_tags -timeout 120m
=== RUN   TestAccAWSSecurityGroup_tags
=== PAUSE TestAccAWSSecurityGroup_tags
=== CONT  TestAccAWSSecurityGroup_tags
--- PASS: TestAccAWSSecurityGroup_tags (61.98s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	62.034s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSSecurityGroup_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSSecurityGroup_basic -timeout 120m
=== RUN   TestAccAWSSecurityGroup_basic
=== PAUSE TestAccAWSSecurityGroup_basic
=== CONT  TestAccAWSSecurityGroup_basic
--- PASS: TestAccAWSSecurityGroup_basic (30.05s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	30.108s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSNetworkAcl_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSNetworkAcl_basic -timeout 120m
=== RUN   TestAccAWSNetworkAcl_basic
=== PAUSE TestAccAWSNetworkAcl_basic
=== CONT  TestAccAWSNetworkAcl_basic
--- PASS: TestAccAWSNetworkAcl_basic (27.33s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	27.389s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSNetworkAcl_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSNetworkAcl_tags -timeout 120m
=== RUN   TestAccAWSNetworkAcl_tags
=== PAUSE TestAccAWSNetworkAcl_tags
=== CONT  TestAccAWSNetworkAcl_tags
--- PASS: TestAccAWSNetworkAcl_tags (56.40s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	56.461s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSInternetGateway_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSInternetGateway_basic -timeout 120m
=== RUN   TestAccAWSInternetGateway_basic
=== PAUSE TestAccAWSInternetGateway_basic
=== CONT  TestAccAWSInternetGateway_basic
--- PASS: TestAccAWSInternetGateway_basic (69.84s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	69.897s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSInternetGateway_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSInternetGateway_tags -timeout 120m
=== RUN   TestAccAWSInternetGateway_tags
=== PAUSE TestAccAWSInternetGateway_tags
=== CONT  TestAccAWSInternetGateway_tags
--- PASS: TestAccAWSInternetGateway_tags (68.57s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	68.629s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSEgressOnlyInternetGateway_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSEgressOnlyInternetGateway_basic -timeout 120m
=== RUN   TestAccAWSEgressOnlyInternetGateway_basic
=== PAUSE TestAccAWSEgressOnlyInternetGateway_basic
=== CONT  TestAccAWSEgressOnlyInternetGateway_basic
--- PASS: TestAccAWSEgressOnlyInternetGateway_basic (27.02s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	27.074s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSEgressOnlyInternetGateway_Tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSEgressOnlyInternetGateway_Tags -timeout 120m
=== RUN   TestAccAWSEgressOnlyInternetGateway_Tags
=== PAUSE TestAccAWSEgressOnlyInternetGateway_Tags
=== CONT  TestAccAWSEgressOnlyInternetGateway_Tags
--- PASS: TestAccAWSEgressOnlyInternetGateway_Tags (57.39s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	57.442s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSVPCPeeringConnection_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSVPCPeeringConnection_basic -timeout 120m
=== RUN   TestAccAWSVPCPeeringConnection_basic
=== PAUSE TestAccAWSVPCPeeringConnection_basic
=== CONT  TestAccAWSVPCPeeringConnection_basic
--- PASS: TestAccAWSVPCPeeringConnection_basic (28.58s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	28.640s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSVPCPeeringConnection_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSVPCPeeringConnection_tags -timeout 120m
=== RUN   TestAccAWSVPCPeeringConnection_tags
=== PAUSE TestAccAWSVPCPeeringConnection_tags
=== CONT  TestAccAWSVPCPeeringConnection_tags
--- PASS: TestAccAWSVPCPeeringConnection_tags (61.71s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	61.760s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSENI_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSENI_tags -timeout 120m
=== RUN   TestAccAWSENI_tags
=== PAUSE TestAccAWSENI_tags
=== CONT  TestAccAWSENI_tags
--- PASS: TestAccAWSENI_tags (71.93s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	71.987s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSENI_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSENI_basic -timeout 120m
=== RUN   TestAccAWSENI_basic
=== PAUSE TestAccAWSENI_basic
=== CONT  TestAccAWSENI_basic
--- PASS: TestAccAWSENI_basic (37.29s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	37.347s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSSubnet_basic
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSSubnet_basic -timeout 120m
=== RUN   TestAccAWSSubnet_basic
=== PAUSE TestAccAWSSubnet_basic
=== CONT  TestAccAWSSubnet_basic
--- PASS: TestAccAWSSubnet_basic (27.76s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	27.813s
+ make testacc TEST=./aws TESTARGS=-run=TestAccAWSSubnet_tags
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSSubnet_tags -timeout 120m
=== RUN   TestAccAWSSubnet_tags
=== PAUSE TestAccAWSSubnet_tags
=== CONT  TestAccAWSSubnet_tags
--- PASS: TestAccAWSSubnet_tags (58.64s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	58.699s
```

**Comment by DrFaust92** [comment]
Rebased, ill add testAccAvailableAZsNoOptInConfig() if #14500 is approved before this,

**Comment by ewbankkit** [comment]
@DrFaust92 `testAccAvailableAZsNoOptInConfig()` is now available with recently merged #14568.

**Comment by DrFaust92** [comment]
Rebased once more + added AZ func use

**Comment by DrFaust92** [comment]
Rebased for test config changes

**Comment by ewbankkit** [comment]
@DrFaust92 Could you add something to the [`aws_spot_instance_request` documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/spot_instance_request) to call out that `tags` applies to the spot instance request, not any instance that fulfills the request? This is a continued source of confusion (#32, #5207).
Thanks.

**Comment by bflad** [comment]
@DrFaust92 please note that "Closes #" usage in the PR description must be done either on individual lines or with the keyword in front of each issue number

**Comment by bflad** [comment]
(Reference: #14383 as well, just linking via comment)

File: aws/resource_aws_subnet_test.go [hunk 3]
```diff
@@ -438,13 +484,52 @@ resource "aws_vpc" "test" {
 resource "aws_subnet" "test" {
   cidr_block              = "10.1.1.0/24"
   vpc_id                  = aws_vpc.test.id
-  map_public_ip_on_launch = true
+	map_public_ip_on_launch = true
```
Comment (bflad) [inline aws/resource_aws_subnet_test.go:487]: Nit: Whitespace adjustment

```suggestion
  map_public_ip_on_launch = true
```

File: website/docs/r/spot_instance_request.html.markdown [hunk 4]
```diff
@@ -68,6 +68,8 @@ Spot Instance Requests support all the same arguments as
 * `valid_from` - (Optional) The start date and time of the request, in UTC [RFC3339](https://tools.ietf.org/html/rfc3339#section-5.8) format(for example, YYYY-MM-DDTHH:MM:SSZ). The default is to start fulfilling the request immediately.
 * `tags` - (Optional) A map of tags to assign to the resource.
 
+~> **Note:** the `tags` argument applies to the Spot Instance Request itself, not any launched instance.
+
```
Comment (bflad) [inline website/docs/r/spot_instance_request.html.markdown:72]: Note: In the future where the documentation is automatically generated, these types of notes will need to be in the attribute/argument description unless they are worthy of being part of the resource description.

```suggestion
* `tags` - (Optional) A map of tags to assign to the Spot Instance Request. These tags are not automatically applied to the launched Instance.
```

**Pull request merged**